### PR TITLE
fix Issue 14782 - Internal error: backend/cod1.c

### PR DESCRIPTION
**Pull request by WalterBright**
https://issues.dlang.org/show_bug.cgi?id=14782

### CODE REVIEW

File: src/backend/cgelem.c [hunk 1]
```diff
@@ -3170,7 +3180,10 @@ elem * elstruct(elem *e, goal_t goal)
                     goto L1;
                 }
             }
-            if (I64 && (ty == TYstruct || (ty == TYarray && config.exe == EX_WIN64)))
+            if (I64 && (ty == TYstruct ||
+                       (ty == TYarray && config.exe == EX_WIN64) ||
+                       (0 && ty == TYarray && !tysimd(e->Ety)))
```
Comment (DmitryOlshansky) [inline src/backend/cgelem.c]: remove that `0 &&` ? Git has full history anyway no need for dead code to stay around.

File: test/runnable/mars1.d [hunk 2]
```diff
@@ -1357,6 +1357,45 @@ void test2()
...
+
+// 14782
+
+
+void test14782()
+{
+    static struct Foo
+    {
+	long a = 8;
+	int b = 7;
```
Comment (9rnsr) [inline test/runnable/mars1.d]: tab -> spaces

File: test/runnable/mars1.d [hunk 3]
```diff
@@ -1357,6 +1357,45 @@ void test2()
...
+    assert(result[0].b == 7);
+}
+
+////////////////////////////////////////////////////////////////////////
+
+void test14987()
+{
+    static struct Foo
+    {
+	int b = 7;
```
Comment (9rnsr) [inline test/runnable/mars1.d]: tab -> spaces

File: test/runnable/mars1.d [hunk 4]
```diff
@@ -20,12 +20,12 @@ int testswitch()
     i = 3;
     switch (i)
     {
-	case 0:
-	case 1:
-	default:
-	    assert(0);
-	case 3:
-	    break;
+        case 0:
```
Comment (andralex) [inline test/runnable/mars1.d]: aren't case labels flush with the switch?

**Comment by andralex** [comment]
Auto-merge toggled on

File: test/runnable/mars1.d (hunk 4 above)
Comment (WalterBright) [inline test/runnable/mars1.d]: Not a style I've ever used.

**Pull request merged**